lib/elixir/lib/calendar.ex

```diff
@@ -1214,10 +1214,15 @@ defmodule NaiveDateTime do
   def to_iso8601(naive_datetime)
 
   def to_iso8601(%{year: year, month: month, day: day,
-                   hour: hour, minute: minute, second: second, microsecond: microsecond}) do
+                   hour: hour, minute: minute, second: second, microsecond: microsecond, calendar: Calendar.ISO}) do
     Calendar.ISO.naive_datetime_to_iso8601(year, month, day, hour, minute, second, microsecond)
   end
 
+  def to_iso8601(%{year: _, month: _, day: _,
+                   hour: _, minute: _, second: _, microsecond: _, calendar: _} = naive_datetime) do
+    raise ArgumentError, "cannot convert #{inspect naive_datetime} to the ISO8601 format, because it does not use Calendar.ISO."
```

Comment:
Error messages should not have trailing.
Also let's use `ISO 8601` wording. :bowtie: